pkg/downloader/chart_downloader.go

```diff
@@ -141,36 +142,41 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+var sha256re = regexp.MustCompile(`^sha256:[0-9a-fA-F]{64}$`)
+
 func (c *ChartDownloader) getOciURI(ref, version string, u *url.URL) (*url.URL, error) {
-	var tag string
 	var err error
 
 	// Evaluate whether an explicit version has been provided. Otherwise, determine version to use
-	_, errSemVer := semver.NewVersion(version)
-	if errSemVer == nil {
-		tag = version
+	if sha256re.MatchString(version) {
+		u.Path = fmt.Sprintf("%s@%s", u.Path, version)
```

Comment:
This lgtm in general. My only question is why all of the existing code needs to move around for this. Can we just move this check to the top and return if true, leaving the rest of the code as-is?

Reply:
The rest of the code is effectively untouched. The large diff seems to be generated by the move of `var tag string` to the `else` scope as this is the only place where it is used. The diff matches the wrong `} else {`.

This wraps the existing logic with a condition `if sha256re...` that sets `u.Path` otherwise the previous logic is run unmodified and sets `u.Path` as it previously did.

Reply:
Github seems to support it too. Click the cog at the top of the `Files Changed` tab and click `Hide whitespaces`